models/Dependency/Dao.php

```diff
@@ -196,6 +197,64 @@ public function getRequiredBy($offset = null, $limit = null)
         return $requiredBy;
     }
 
+    /**
+     * @param null $orderBy
+     * @param null $orderDirection
+     * @param null $offset
+     * @param null $limit
+     * @return array
+     */
+    public function getRequiredByWithPath($orderBy = null, $orderDirection = null, $offset = null, $limit = null)
+    {
+        $targetId = intval($this->model->getSourceId());
+
+        if (in_array($this->model->getSourceType(), ['object', 'document', 'asset'])) {
+            $targetType = $this->model->getSourceType();
+        } else {
+            throw new SuspiciousOperationException('Illegal source type ' . $this->model->getSourceType());
+        }
+
+        if (!in_array($orderBy, ['id', 'type', 'path'])) {
+            $orderBy = 'id';
+        }
+
+        if (!in_array($orderDirection, ['ASC', 'DESC'])) {
+            $orderDirection = 'ASC';
+        }
+
+        $query = "
+            SELECT id, type, path
```

Comment:
@t-zilla can you please explain what is the benefit of using such a complex query over a simple one in `getRequiredBy()`? AFAIK we still need to fetch the relation elements to check permissions and so.. or i missed a point here?

Reply:
To enable pagination without a need to fetch _all_ dependencies which causes a communication error down the line. The simple query on `dependencies` table doesn't enable us to sort by `path` which appears on the S&R grid.

Reply:
Thanks for your explanation! I believe we can keep it then as it is just used in S&R grid.